### PR TITLE
V2 model additions

### PR DESCRIPTION
**Pull request by jimnarey**
Add an owner (User) field to application as well as datetimes for when submitted and decided.

### CODE REVIEW

File: request_a_govuk_domain/request/models/application.py [hunk 1]
```diff
@@ -23,6 +24,9 @@ class Application(models.Model):
 
     id = models.BigAutoField(primary_key=True)
     reference = models.CharField(max_length=REF_NUM_LENGTH, null=False)
+    owner = models.ForeignKey(User, on_delete=models.CASCADE, null=True)
+    when_submitted = models.DateTimeField(auto_now_add=True)
```
Comment (maxf) [inline request_a_govuk_domain/request/models/application.py:28]: I think `date_submitted`, or just `submitted` is clearer. To me `when_submitted` sounds like the name of an action like `on_delete`. Also maybe you can add a description field in the definition.

File: request_a_govuk_domain/request/migrations/0001_initial.py [hunk 2]
```diff
@@ -34,6 +34,14 @@ class Migration(migrations.Migration):
                 ),
                 ("domain_name", models.CharField(max_length=253)),
                 ("written_permission_evidence", models.FileField(upload_to="")),
+                (
+                    "owner",
```
Comment (maxf) [inline request_a_govuk_domain/request/migrations/0001_initial.py:38]: There's no model called "owner". Is that still valid?

**Review by maxf — COMMENTED** [review]
Looks like the test errors is due to a bootstrap issue but I've not looked in detail

File: request_a_govuk_domain/request/models/application.py [hunk 3]
```diff
@@ -23,6 +24,9 @@ class Application(models.Model):
 
     id = models.BigAutoField(primary_key=True)
     reference = models.CharField(max_length=REF_NUM_LENGTH, null=False)
+    owner = models.ForeignKey(User, on_delete=models.CASCADE, null=True)
```
Comment (prakash-gds) [inline request_a_govuk_domain/request/models/application.py:27]: Regarding on_delete, Do we want to delete the application when User is deleted? Would SET_NULL be a better option, considering the field is nullable?

**Review by prakash-gds — APPROVED** [review]

**Pull request merged**